src/Distributed/distributed_grids.jl

```diff
@@ -271,15 +271,41 @@ function with_halo(new_halo, grid::DistributedImmersedBoundaryGrid)
     global_immmersed_grid = reconstruct_global_grid(grid)
     immersed_boundary     = global_immmersed_grid.immersed_boundary
     underlying_grid       = with_halo(new_halo, global_immmersed_grid)
-    immersed_boundary     = resize_immersed_boundary!(immersed_boundary, underlying_grid)
+    immersed_boundary     = resize_immersed_boundary(immersed_boundary, underlying_grid)
     new_grid              = ImmersedBoundaryGrid(underlying_grid, immersed_boundary)
-
     return scatter_local_grids(architecture(grid), new_grid)
 end
 
-resize_immersed_boundary!(ib::AbstractGridFittedBottom{<:Function}, grid) = ib
+using Oceananigans.ImmersedBoundaries: AbstractGridFittedBottom, GridFittedBoundary, compute_mask
+
+"""
+    function resize_immersed_boundary!(ib, grid)
+
+If the immersed condition is an `OffsetArray`, resize it to match the 
+data size associated with the `grid`
+"""
+resize_immersed_boundary(ib::AbstractGridFittedBottom, grid) = ib
+resize_immersed_boundary(ib::GridFittedBoundary, grid)       = ib
+
+function resize_immersed_boundary(ib::GridFittedBoundary{<:OffsetArray}, grid)
+
+    Nx, Ny, Nz = size(grid)
+    Hx, Hy, Nz = halo_size(grid)
+
+    mask_size = (Nx, Ny, Nz) .+ 2 .* (Hx, Hy, Hz)
```

Comment:
Is this the same as `total_size(grid)`?

Reply:
I think it's important to use the functions here. Remember you want to change the size of a `Center, Bounded` field. When you do that, the above code is wrong.

Reply:
I think you want `total_size((Center, Center, Nothing), grid)` via

https://github.com/CliMA/Oceananigans.jl/blob/e394bf7ef7e4341ffddd449f52f58cee84698813/src/Grids/grid_utils.jl#L84